internal/services/cdn/cdn_frontdoor_route_resource.go

```diff
@@ -205,33 +222,69 @@ func resourceCdnFrontDoorRouteCreate(d *pluginsdk.ResourceData, meta interface{}
 		return tf.ImportAsExistsError("azurerm_cdn_frontdoor_route", id.ID())
 	}
 
+	var origins []interface{}
+	var originGroup *cdn.ResourceReference
+
 	protocolsRaw := d.Get("supported_protocols").(*pluginsdk.Set).List()
+	originGroupRaw := d.Get("cdn_frontdoor_origin_group_id").(string)
+	ruleSetIdsRaw := d.Get("cdn_frontdoor_rule_set_ids").([]interface{})
+	originsRaw := d.Get("cdn_frontdoor_origin_ids").([]interface{})
+	customDomainsRaw := d.Get("cdn_frontdoor_custom_domain_ids").([]interface{})
 	httpsRedirect := d.Get("https_redirect_enabled").(bool)
+	linkToDefaultDomain := d.Get("link_to_default_domain").(bool)
 
+	// NOTE: If HTTPS Redirect is enabled the Supported Protocols must support both HTTP and HTTPS
+	// This configuration does not cause an error when provisioned, however the http requests that
+	// are supposed to be redirected to https remain http requests
 	if httpsRedirect {
-		// NOTE: If HTTPS Redirect is enabled the Supported Protocols must support both HTTP and HTTPS
-		// This configuration does not cause an error when provisioned, however the http requests that
-		// are supposed to be redirected to https remain http requests
-		err := validate.SupportsBothHttpAndHttps(protocolsRaw, "https_redirect_enabled")
+		if err := validate.SupportsBothHttpAndHttps(protocolsRaw, "https_redirect_enabled"); err != nil {
+			return err
+		}
+	}
+
+	normalizedCustomDomains, err := customDomainsInsensitively(customDomainsRaw)
+	if err != nil {
+		return err
+	}
+
+	if !linkToDefaultDomain && len(normalizedCustomDomains) == 0 {
+		return fmt.Errorf("it is invalid to disable the 'LinkToDefaultDomain' for the CDN Front Door Route(Name: %s) since the route does not have any CDN Front Door Custom Domains associated with it", id.RouteName)
+	} else if len(normalizedCustomDomains) != 0 {
+		if err := sliceHasDuplicates(normalizedCustomDomains, "CDN FrontDoor Custom Domain"); err != nil {
+			return err
+		}
+
+		if err := validateRoutesCustomDomainProfile(normalizedCustomDomains, id.RouteName, id.ProfileName); err != nil {
+			return err
+		}
+	}
+
+	if originGroupRaw != "" {
+		id, err := parse.FrontDoorOriginGroupIDInsensitively(originGroupRaw)
```

Comment:
this is an expand function where this is coming from the config, so we're validating the casing:

```suggestion
		id, err := parse.FrontDoorOriginGroupID(originGroupRaw)
```

Reply:
Fixed.